Source/DafnyCore/Backends/SinglePassCodeGenerator/SinglePassCodeGenerator.Statement.cs

```diff
@@ -180,14 +180,67 @@ protected void TrStmt(Statement stmt, ConcreteSyntaxTree wr, ConcreteSyntaxTree
             break;
           }
         case ExpectStmt expectStmt: {
-            var s = expectStmt;
             // TODO there's potential here to use target-language specific features such as exceptions
             // to make it more target-language idiomatic and improve performance
-            ConcreteSyntaxTree bodyWriter = EmitIf(out var guardWriter, false, wr);
-            var negated = new UnaryOpExpr(s.Tok, UnaryOpExpr.Opcode.Not, s.Expr);
-            negated.Type = Type.Bool;
-            EmitExpr(negated, false, guardWriter, wStmts);
-            EmitHalt(s.Tok, s.Message, bodyWriter);
+            // For now, this code prints nicely only in the Rust code generator until we make it work for every code generator
+            var specialExpectEqualHandling = Options.Backend.TargetId == "rs";
```

Comment:
Happy with this for now, but can you cut an issue linked to https://github.com/dafny-lang/dafny/issues/5561 to track this? At a minimum I'd like to tweak the output format a little too.

Reply:
Here is the issue.
https://github.com/dafny-lang/dafny/issues/5720
However, I'm not sure I should track it to the Dafny-to-Rust code generator since it's not related.